Issue In Siddhi store RDBMS

I am getting this error when i send an event to SP 4.3.0 M6.

 [2018-08-28 17:09:47,331] ERROR {org.wso2.siddhi.core.query.processor.stream.AbstractStreamProcessor} - Error on 'IS_ANALYTICS_IDLE_SESSION_TERMINATION'. Error retrieving records from table 'ActiveSessionsTable': Invalid value "2" for parameter "parameterIndex" [90008-196] Dropping event chunk EventChunk{first=null}, error in processing org.wso2.siddhi.core.query.processor.stream.window.LengthWindowProcessor. org.wso2.extension.siddhi.store.rdbms.exception.RDBMSTableException: Error on 'IS_ANALYTICS_IDLE_SESSION_TERMINATION'. Error retrieving records from table 'ActiveSessionsTable': Invalid value "2" for parameter "parameterIndex" [90008-196]
	at org.wso2.extension.siddhi.store.rdbms.RDBMSEventTable.find(RDBMSEventTable.java:485)
	at org.wso2.siddhi.core.table.record.AbstractRecordTable.find(AbstractRecordTable.java:129)
	at org.wso2.siddhi.core.table.Table.find(Table.java:173)
	at org.wso2.siddhi.core.query.processor.stream.window.TableWindowProcessor.find(TableWindowProcessor.java:67)
	at org.wso2.siddhi.core.query.input.stream.join.JoinProcessor.process(JoinProcessor.java:85)
	at org.wso2.siddhi.core.query.processor.stream.window.LengthWindowProcessor.process(LengthWindowProcessor.java:137)
	at org.wso2.siddhi.core.query.processor.stream.window.WindowProcessor.processEventChunk(WindowProcessor.java:66)
	at org.wso2.siddhi.core.query.processor.stream.AbstractStreamProcessor.process(AbstractStreamProcessor.java:125)
	at org.wso2.siddhi.core.query.input.stream.join.JoinProcessor.process(JoinProcessor.java:118)
	at org.wso2.siddhi.core.query.input.ProcessStreamReceiver.processAndClear(ProcessStreamReceiver.java:186)
	at org.wso2.siddhi.core.query.input.ProcessStreamReceiver.process(ProcessStreamReceiver.java:96)
	at org.wso2.siddhi.core.query.input.ProcessStreamReceiver.receive(ProcessStreamReceiver.java:132)
	at org.wso2.siddhi.core.stream.StreamJunction.sendEvent(StreamJunction.java:179)
	at org.wso2.siddhi.core.trigger.PeriodicTrigger$1.run(PeriodicTrigger.java:80)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.h2.jdbc.JdbcSQLException: Invalid value "2" for parameter "parameterIndex" [90008-196]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.getInvalidValueException(DbException.java:228)
	at org.h2.jdbc.JdbcPreparedStatement.setParameter(JdbcPreparedStatement.java:1444)
	at org.h2.jdbc.JdbcPreparedStatement.setInt(JdbcPreparedStatement.java:343)
	at com.zaxxer.hikari.proxy.HikariPreparedStatementProxy.setInt(HikariPreparedStatementProxy.java)
	at org.wso2.extension.siddhi.store.rdbms.util.RDBMSTableUtils.populateStatementWithSingleElement(RDBMSTableUtils.java:259)
	at org.wso2.extension.siddhi.store.rdbms.util.RDBMSTableUtils.resolveCondition(RDBMSTableUtils.java:172)
	at org.wso2.extension.siddhi.store.rdbms.RDBMSEventTable.find(RDBMSEventTable.java:475)
	... 20 more

--- a/component/src/main/java/org/wso2/extension/siddhi/store/rdbms/RDBMSEventTable.java
+++ b/component/src/main/java/org/wso2/extension/siddhi/store/rdbms/RDBMSEventTable.java
@@ -400,7 +400,6 @@ public class RDBMSEventTable extends AbstractRecordTable {
     private String stringType;
     private String stringSize;
     private String recordContainsConditionTemplate;
-    private String findCondition;
 
     @Override
     protected void init(TableDefinition tableDefinition, ConfigReader configReader) {
@@ -439,13 +438,12 @@ public class RDBMSEventTable extends AbstractRecordTable {
     protected RecordIterator<Object[]> find(Map<String, Object> findConditionParameterMap,
                                             CompiledCondition compiledCondition) {
         RDBMSCompiledCondition rdbmsCompiledCondition = (RDBMSCompiledCondition) compiledCondition;
-        if (findCondition == null) {
-            if (rdbmsCompiledCondition.isContainsConditionExist()) {
-                findCondition = RDBMSTableUtils.processFindConditionWithContainsConditionTemplate(
-                        rdbmsCompiledCondition.getCompiledQuery(), this.recordContainsConditionTemplate);
-            } else {
-                findCondition = rdbmsCompiledCondition.getCompiledQuery();
-            }
+        String findCondition;
+        if (rdbmsCompiledCondition.isContainsConditionExist()) {
+            findCondition = RDBMSTableUtils.processFindConditionWithContainsConditionTemplate(
+                    rdbmsCompiledCondition.getCompiledQuery(), this.recordContainsConditionTemplate);
+        } else {
+            findCondition = rdbmsCompiledCondition.getCompiledQuery();
         }
         //Some databases does not support single condition on where clause.
         //(atomic condition on where clause: SELECT * FROM TABLE WHERE true)

